src/pipeline/pipeline.cpp

```diff
@@ -199,35 +200,24 @@ namespace librealsense
             return _streams_to_sync_ids;
         }
 
-        frame_callback_ptr pipeline::get_callback(std::vector<int> synced_streams_ids)
+        rs2_frame_callback_sptr pipeline::get_callback(std::vector<int> synced_streams_ids)
         {
-            auto pipeline_process_callback = [&](frame_holder fref)
-            {
-                _aggregator->invoke(std::move(fref));
-            };
-
-            frame_callback_ptr to_pipeline_process = {
-                new internal_frame_callback<decltype(pipeline_process_callback)>(pipeline_process_callback),
-                [](rs2_frame_callback* p) { p->release(); }
-            };
-
-            _syncer->set_output_callback(to_pipeline_process);
+            _syncer->set_output_callback(
+                make_frame_callback( [&]( frame_holder fref ) { _aggregator->invoke( std::move( fref ) ); } ) );
 
-            auto to_syncer = [&, synced_streams_ids](frame_holder fref)
-            {
-                // if the user requested to sync the frame push it to the syncer, otherwise push it to the aggregator
-                if (std::find(synced_streams_ids.begin(), synced_streams_ids.end(), fref->get_stream()->get_unique_id()) != synced_streams_ids.end())
-                    _syncer->invoke(std::move(fref));
-                else
-                    _aggregator->invoke(std::move(fref));
-            };
-
-            frame_callback_ptr rv = {
-                new internal_frame_callback<decltype(to_syncer)>(to_syncer),
-                [](rs2_frame_callback* p) { p->release(); }
```

Comment:
No need for this release anymore?
Just need to make sure no memory leak here

Reply:
It's still there: `make_frame_callback` adds it automatically.